node/impl/client/client.go

```diff
@@ -1295,16 +1278,20 @@ func (a *API) ClientCalcCommP(ctx context.Context, inpath string) (*api.CommPRet
 		return nil, xerrors.Errorf("seek to start: %w", err)
 	}
 
-	pieceReader, pieceSize := padreader.New(rdr, uint64(stat.Size()))
-	commP, err := ffiwrapper.GeneratePieceCIDFromFile(arbitraryProofType, pieceReader, pieceSize)
```

Comment:
This kind of makes sense but I don't understand why this ffiwrapper thing exists and was called in the first place if the commp writer thing could just do a direct sum over the piece bytes in the first place.

Reply:
The writer thing is relatively recent, and this API existed for quite some time before.

Reply:
(the writer also calls FFI, it just does that from multiple threads for smaller chunks, then creates the top tree from them)